Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp

```diff
@@ -3559,51 +3604,123 @@ static std::optional<CSSGradientColorStopList> consumeGradientColorStops(CSSPars
     return { WTFMove(stops) };
 }
 
+static std::optional<CSSGradientColorStopList> consumeLengthColorStopList(CSSParserTokenRange& range, const CSSParserContext& context, SupportsColorHints supportsColorHints)
+{
+    return consumeColorStopList(range, context, supportsColorHints, [&] { return consumeLengthOrPercent(range, context.mode, ValueRange::All); });
+}
+
+static std::optional<CSSGradientColorStopList> consumeAngularColorStopList(CSSParserTokenRange& range, const CSSParserContext& context, SupportsColorHints supportsColorHints)
+{
+    return consumeColorStopList(range, context, supportsColorHints, [&] { return consumeAngleOrPercent(range, context.mode); });
+}
+
 static RefPtr<CSSValue> consumePrefixedRadialGradient(CSSParserTokenRange& args, const CSSParserContext& context, CSSGradientRepeat repeating)
 {
-    auto centerCoordinate = consumeOneOrTwoValuedPositionCoordinates(args, context.mode, UnitlessQuirk::Forbid);
-    if (centerCoordinate && !consumeCommaIncludingWhitespace(args))
-        return nullptr;
+    // https://compat.spec.whatwg.org/#css-gradients-webkit-radial-gradient/ states that -webkit-radial-gradient() and
+    // -webkit-repeating-radial-gradient() must be "treated as an alias of radial-gradient as defined in [css3-images-20110217]."
+    // In [css3-images-20110217] the grammar was defined as:
+    //
+    //   <radial-gradient> = radial-gradient([<bg-position>,]? [ [<shape> || <size>] | [<length> | <percentage>]{2} ,]? <color-stop>[, <color-stop>]+);
+    //   <shape> = [ circle | ellipse ]
+    //   <size> = [ closest-side | closest-corner | farthest-side | farthest-corner | contain | cover ]
+    //
+    //      defaults to ‘ellipse cover’.
+    //
+    // see https://www.w3.org/TR/2011/WD-css3-images-20110217/#radial-gradients.
+
+    std::optional<CSSGradientPosition> position;
+    if (auto centerCoordinate = consumeOneOrTwoValuedPositionCoordinates(args, context.mode, UnitlessQuirk::Forbid)) {
+        if (!consumeCommaIncludingWhitespace(args))
+            return nullptr;
+        position = CSSGradientPosition { WTFMove(centerCoordinate->x), WTFMove(centerCoordinate->y) };
+    }
+
+    auto consumeShapeKeyword = [](auto& args) -> std::optional<CSSPrefixedRadialGradientValue::ShapeKeyword> {
+        switch (args.peek().id()) {
+        case CSSValueCircle:
+            args.consumeIncludingWhitespace();
+            return CSSPrefixedRadialGradientValue::ShapeKeyword::Circle;
+        case CSSValueEllipse:
+            args.consumeIncludingWhitespace();
+            return CSSPrefixedRadialGradientValue::ShapeKeyword::Ellipse;
+        default:
+            return std::nullopt;
+        }
+    };
+
+    auto consumeExtentKeyword = [](auto& args) -> std::optional<CSSPrefixedRadialGradientValue::ExtentKeyword> {
+        switch (args.peek().id()) {
+        case CSSValueClosestSide:
+            args.consumeIncludingWhitespace();
```

Comment:
So many different idioms for consuming expected identifiers! This one is fine.

Reply:
Yeah, going to make a template function to fix this. https://bugs.webkit.org/show_bug.cgi?id=247326. I think I can use SortedArrayMap to make it nice.